src/main/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTablePresentation.java

```diff
@@ -358,13 +358,8 @@ private static Object getValueAtDetailAST(DetailAST ast, int column) {
      * @return DetailNode tree
      */
     private DetailNode getJavadocTree(DetailAST blockComment) {
-        DetailNode javadocTree = blockCommentToJavadocTree.get(blockComment);
-        if (javadocTree == null) {
-            javadocTree = new JavadocDetailNodeParser().parseJavadocAsDetailNode(blockComment)
-                    .getTree();
-            blockCommentToJavadocTree.put(blockComment, javadocTree);
-        }
-        return javadocTree;
+        return blockCommentToJavadocTree.computeIfAbsent(blockComment,
+            key -> new JavadocDetailNodeParser().parseJavadocAsDetailNode(blockComment).getTree());
```

Comment:
I might be old-school, but I like to have single return statement, so it is easy to put break point on return and validate result. Extra local variable is super easy task for any optimizer, but it improve code maintenance.

@rnveach , what is your position on this ?

Reply:
I am even more old school then you. I am not a fan of lambdas and the new streams in Java 8 because they hide some functionality and some of them make the code harder to read, imo. I maintain the backport and these always give me issues converting since I have to understand them to convert them. Understanding them never comes instant for me even though I have converted a bunch.

Can't say I have really tried to put a breakpoint here but I tried and Eclipse Oxygen won't let me put it on the `key ->` line unless there was {}s. Didn't try newer ones.
New code is still easy to read. We have alot of other areas I would choose to remove lambdas first over starting with this.
I am pretty much fine with this one.

Reply:
@strkkk ,

Please do me favor, keep `return javadocTree`.

Reply:
@romani Ok, I've brought this one back.